developers/plugin-architecture/write-custom-plugins/README.md

```diff
@@ -1,7 +1,53 @@
 # Write Custom Plugins
 
-Pinot allows user to easily create and use new plugins to support custom needs. The developer needs to implement a few standard interfaces specific to the the type of plugins in Java.
+Pinot allows user to easily create and use new plugins to support custom needs. The developer needs to implement a few standard interfaces specific to the type of plugins in Java.
 
-Once the code is ready, you can put the complete JAR file in pinot `/plugins` directory. All the classes in plugins directory are loaded at pinot's startup.\
-\
-We also encourage users to shaded commonly used dependencies such as `guava`, `jackson`&#x20;
+Once the code is ready, you can put the complete JAR file in pinot `/plugins` directory. All the classes in plugins directory are loaded at pinot's startup.
+
+As of Apache Pinot 1.3.0 there's an additional support for a new structure:
+
+     /plugins
+        /my-custom-plugin
+           /classes/...
+           /com-2.3.1.jar
+           /foo-1.2.1.jar
+           /bar-3.1.1.jar
+           /pinot-plugin.properties
+
+This way there's no need to shade classes anymore, because Pinot will ensure isolation and proper ordered loading of these classes. 
+Pinot will only expose the classes in pinot-spi.
+The `classes`-directory will contain custom compiled Java-code of the plugin.
+Next to these `classes` are all the runtime required dependencies, but without the pinot-jars.
+The `pinot-plugin.properties` is currently a placeholder to recognize this new structure. In the future this file will be used to store metadata for the plugin.
```

Comment:
Could we explain this properties here as well? I know that PR is not merged yet, but this documentation won't be merged before that PR anyway.

Reply:
you mean the current `parent.realmId=pinot`?

Reply:
That and importFrom